pandas_dev_flaker/_ast_helpers.py

```diff
@@ -37,13 +37,40 @@ def is_str_constant(
     node: ast.Call,
 ) -> bool:
     return isinstance(node.func, ast.Attribute) and (
-        (
-            sys.version_info < (3, 8)
-            and isinstance(node.func.value, ast.Str)
-        )
+        (sys.version_info < (3, 8) and isinstance(node.func.value, ast.Str))
         or (
             sys.version_info >= (3, 8)
             and isinstance(node.func.value, ast.Constant)
             and isinstance(node.func.value.value, str)
         )
     )
+
+
+def parse_annotation(node: ast.BinOp) -> bool:
+    series, any_array_like = "Series", "AnyArrayLike"
+    is_series, is_array_like = False, False
+
+    while isinstance(node, ast.BinOp):
+        if isinstance(node.right, ast.Name):
+            if node.right.id == series:
+                is_series = True
+            if node.right.id == any_array_like:
+                is_array_like = True
+        else:
+            raise AssertionError
```

Comment:
wouldn't a test case like
```
        pytest.param(
            "def p(foo, other: Series | Union[int, str] ): pass",
            id="Function argument with two annotations",
        ),
```
hit this?

Reply:
this test worked. i was expecting the test i wrote on line 31 in `disallow_argument_types.py` to have covered such a case